Add TCP stats to deployment dashboards

hawkw opened the pull request:
This PR adds the TCP metrics added in #785 and #790 to the Grafana deployment dashboards. I've added three new graphs in the "Inbound Traffic" and "Outbound Traffic" headings:
+ "TCP Connection Closes Per Second": plots the number of opened and accepted connections that have closed successfully or failed.
+ "Bytes Per Second": plots the number of bytes sent and bytes received on successfully closed TCP connections
+ "TCP Connection Duration": plots the p50, p95, and p99 durations of closed TCP connections

Here's a screenshot of the new graphs, generated using `simulate-proxy` output: 
![screen shot 2018-04-20 at 12 54 08 pm](https://user-images.githubusercontent.com/2796466/39071114-b278ebac-449a-11e8-80e5-0dc25765a618.png)
(note that the unrealistically high latencies/connection durations are just what simulate-proxy currently generates)

I'm planning on adding similar graphs to other dashboards as well in subsequent PRs.

siggy commented on grafana/dashboards/deployment.json:
```diff
@@ -676,6 +947,13 @@
           "id": 39,
           "links": [],
           "mode": "html",
+          "scopedVars": {
```
apologies i forgot to mention you should remove any `scopedVars` blocks.

hawkw commented:
@olix0r 
> Ultimately, I think we need to get clearer on exactly what we want to record...

Yeah, I agree completely. I wasn't sure exactly what graphs we ought to have in the dashboard, so this just represents my best guess. Happy to have further discussions on what should change.

> (because we don't have read and write events.. we probably want to add them but these could come with a real cost).

 I've been wondering about this as well. Maybe, rather than sending an event on _every_ read/write, we might send an occasional "heartbeat" type event so we can get bytes sent/received stats out of the connection with enough resolution to make useful metrics, but without having to handle an event on every read or write. On the other hand, maybe the extra overhead of accounting for whether or not to send an event would be worse than just sending every event?

hawkw commented:
Also @olix0r, I'm considering making the following changes based on your feedback:
+ put TCP charts in a collapsed row
+ rename `BYTES PER SECOND` to `BYTES PER CONNECTION`
+ change `CONNECTIONS CLOSED PER SECOND` to `CONNECTIONS CLOSED` and make it display a total rather than a rate
+ maybe change the `TCP CONNECTION DURATION` chart to a bar graph type histogram rather than a plot of durations over time?

What do you think?

hawkw commented:
@olix0r I've made some additional changes based on your feedback. In particular, I've:
+ put TCP charts in a row that's collapsed by default
+ renamed BYTES PER SECOND to BYTES PER CONNECTION
+ changed CONNECTIONS CLOSED PER SECOND to CONNECTIONS CLOSED and made it display as a heatmap (histogram over time) rather than as a line graph

Please let me know what you think of these changes!

olix0r commented:
@hawkw if it's not inconvenient, can you include a screencap of the new charts (with ~real data ;)?

hawkw commented:
I _just_ tore down Minikube, so it might take a minute, bit yes.
On Tue, Apr 24, 2018 at 4:41 PM Oliver Gould <notifications@github.com>
wrote:

> @hawkw <https://github.com/hawkw> if it's not inconvenient, can you
> include a screencap of the new charts (with ~real data ;)?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/runconduit/conduit/pull/824#issuecomment-384114234>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACqrsqoG1L2WXbZbchd4191OJZBW1BKxks5tr7gvgaJpZM4TeBEu>
> .
>
-- 
Eliza Weisman
Systems Software Engineer
Buoyant, Inc.

hawkw commented:
@olix0r 
![screen shot 2018-04-24 at 5 46 30 pm](https://user-images.githubusercontent.com/2796466/39220693-8646a9e8-47e7-11e8-8d9a-5f93bf21eb3e.png)

Note that the TCP charts are collapsed by default, so when you open the page it looks like this:
![screen shot 2018-04-24 at 5 46 40 pm](https://user-images.githubusercontent.com/2796466/39220701-97606d40-47e7-11e8-91bb-64b42d890077.png)

hawkw commented:
I've added some additional changes based on the changes to metrics @olix0r made in d4d0e579c2dc626ff72f2403505c59efb0fe0379. The dashboard now looks like this:
![screen shot 2018-04-25 at 2 47 55 pm](https://user-images.githubusercontent.com/2796466/39276577-7f5bab30-489e-11e8-9932-a05265fb4361.png)

Hopefully these new charts are much more usable.

Additionally, I've added (collapsed by default) TCP charts to the Conduit Health dashboard as well.
